sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -241,11 +241,21 @@ query
     : ctes? queryNoWith
     ;
 
-insertInto
+insertIntoTable
     : INSERT OVERWRITE TABLE tableIdentifier (partitionSpec (IF NOT EXISTS)?)?
     | INSERT INTO TABLE? tableIdentifier partitionSpec?
     ;
 
+insertOverwriteDirectory
+    : INSERT OVERWRITE LOCAL? DIRECTORY path=STRING rowFormat? createFileFormat?                     #insertOverwriteHiveDir
+    | INSERT OVERWRITE DIRECTORY (path=STRING)? tableProvider (OPTIONS options=tablePropertyList)?   #insertOverwriteDir
```

Comment:
We should also have `LOCAL?` here. This is just for usability. Our current parser always issue a confusing error message. We can issue a better exception in the visit function.

Reply:
ok. added local.